Allow users to accept & configure stats sending

phproberto opened the pull request:
This tries to improve the stats plugin configuration based in user feedback. 

I'll add the details in:
https://github.com/joomla/joomla-cms/pull/8346

phproberto committed:
[imp] allow users to accept & configure stats sending

wilsonge commented:
Rather than a plugin param - I guess we may as well use Joomla's Debug mode? I just did it that way because I was copying code out Nic's update notification plugin blindly :P https://github.com/joomla/joomla-cms/blob/staging/plugins/system/updatenotification/updatenotification.php#L13

wilsonge commented on plugins/system/stats/stats.php:
```diff
@@ -45,91 +76,269 @@ class PlgSystemStats extends JPlugin
+	 */
+	private function isAjaxRequest()
+	{
+		return (isset($_SERVER['HTTP_X_REQUESTED_WITH']) && strtolower($_SERVER['HTTP_X_REQUESTED_WITH']) == 'xmlhttprequest');
```
Please can we use JInput here to grab these variables (better sanity checks) `return strtolower($this->app->input->server->get(''HTTP_X_REQUESTED_WITH', '')) == 'xmlhttprequest';`

phproberto replied:
Ok. I will change it tomorrow. I also want to use ajax always for sending stats so it never affects page loading time

phproberto commented:
If you mean joomla debug mode I don't like that idea. If I'm going to debug
anything I'd have to wait this request to happen. So I think its own
parameter is ok.
On 10 Nov 2015 3:18 am, "George Wilson" notifications@github.com wrote:

> Rather than a plugin param - I guess we may as well run on Joomla Debug
> mode? I just did it that way because I was copying code out Nic's update
> notification plugin blindly :P
> https://github.com/joomla/joomla-cms/blob/staging/plugins/system/updatenotification/updatenotification.php#L13
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/rdeutz/joomla-cms/pull/2#issuecomment-155261709.

rdeutz commented on plugins/system/stats/stats.xml:
```diff
@@ -34,9 +35,39 @@
 					default="https://developer.joomla.org/stats/submit"
 					description="PLG_SYSTEM_STATS_URL_DESC"
-					label="PLG_SYSTEM_STATS_URL_LABEL"					
+					label="PLG_SYSTEM_STATS_URL_LABEL"
```
I think we should make this hardcoded so that nobody can change it, it is too easy to misuse the plugin. (I kwon that is form the original but because we are changing it we can do this also)

phproberto replied:
I agree because if we change the URL system will keep working. I'll do it.

phproberto committed:
[imp] remove stats plugin url setting & move debug to advanced

phproberto committed:
[imp] always send system stats using AJAX

phproberto commented:
PR updated:
- Removed URL setting. Not it uses a class property.
- Moved the debug setting to the Advanced tab.
- Now the server update is always done through AJAX so it never affects page loading time.
- I have switched AJAX requests to raw mode because it's the the right way to do it. Failed AJAX requests returning a 200 is wrong!
